doc/cprover-manual/contracts-assigns.md

```diff
@@ -1,50 +1,144 @@
 [CPROVER Manual TOC](../../)
 
-# Assigns Clause
+# Assigns Clauses
 
-## In Function Contracts
+An _assigns_ clause lets the user to specify which memory locations may be
+assigned to by a function or a loop.
 
-### Syntax
+The set of locations described by the contract is the union of the locations
+specified by each of its assigns clauses, or the empty set of no _assigns_
+clause is specified.
```

Comment:
There is indeed an inference pass for loop contracts in case there isn't a user-specified clause in the loop contract. 
I'll make the distinction. 
We don't have inference for function contracts. Since we ultimately want to have function contracts as standalone entities we wouldn't know what instructions to infer the assigned memory locations from.